src/editors/containers/ProblemEditor/index.jsx

```diff
@@ -2,22 +2,26 @@ import React from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import { Spinner } from '@edx/paragon';
-import { injectIntl } from '@edx/frontend-platform/i18n';
+import { injectIntl, FormattedMessage } from '@edx/frontend-platform/i18n';
 import SelectTypeModal from './components/SelectTypeModal';
 import EditProblemView from './components/EditProblemView';
 import { selectors, thunkActions } from '../../data/redux';
 import { RequestKeys } from '../../data/constants/requests';
+import messages from './messages';
 
 export const ProblemEditor = ({
   onClose,
   // Redux
   problemType,
   blockFinished,
+  blockFailed,
   studioViewFinished,
   blockValue,
   initializeProblemEditor,
   assetsFinished,
 }) => {
+  React.useEffect(() => initializeProblemEditor(blockValue), []);
```

Comment:
Now that I am paying attention to the specifics: I think the reason that the useEffect was put after the return statement is to not call the `initializeProblemEditor` method unless block, studioview, and assets finished. The comment says "once data is loaded, init store". It was just the wrong approach. How to call the `initializeProblemEditor` method  conditionally is to add the condition to the callback method inside of the useEffect. However, for the hook to register that the booleans have changed, you also need to put them into the dependency array.